src/test/java/core/basesyntax/service/impl/TransactionParserImplTest.java

```diff
@@ -0,0 +1,83 @@
+        fruitTransactionList.add(fruitTransaction);
+        List<String> list = List.of("type;fruit;quantity", "b;banana;20");
+        TransactionParser transactionParser = new TransactionParserImpl();
+        Assert.assertEquals(fruitTransactionList, transactionParser.parse(list));
+    }
+
+    @Test
+    public void parse_SupplyBalance_Ok() {
+        FruitTransaction fruitTransaction = new FruitTransaction();
+        fruitTransaction.setOperation(FruitTransaction.Operation.SUPPLY);
+        fruitTransaction.setFruit("banana");
+        fruitTransaction.setQuantity(100);
+        List<FruitTransaction> fruitTransactionList = new ArrayList<>();
+        fruitTransactionList.add(fruitTransaction);
+        List<String> list = List.of("type;fruit;quantity", "s;banana;100");
+        TransactionParser transactionParser = new TransactionParserImpl();
+        Assert.assertEquals(fruitTransactionList, transactionParser.parse(list));
+    }
+
+    @Test
+    public void parse_PurchaseBalance_Ok() {
+        FruitTransaction fruitTransaction = new FruitTransaction();
+        fruitTransaction.setOperation(FruitTransaction.Operation.PURCHASE);
+        fruitTransaction.setFruit("banana");
+        fruitTransaction.setQuantity(13);
+        List<FruitTransaction> fruitTransactionList = new ArrayList<>();
+        fruitTransactionList.add(fruitTransaction);
+        TransactionParser transactionParser = new TransactionParserImpl();
+        List<String> list = List.of("type;fruit;quantity", "p;banana;13");
+        Assert.assertEquals(fruitTransactionList, transactionParser.parse(list));
+    }
+
+    @Test
+    public void parse_ReturnBalance_Ok() {
+        FruitTransaction fruitTransaction = new FruitTransaction();
+        fruitTransaction.setOperation(FruitTransaction.Operation.RETURN);
+        fruitTransaction.setFruit("apple");
+        fruitTransaction.setQuantity(10);
+        TransactionParser transactionParser = new TransactionParserImpl();
+        List<FruitTransaction> fruitTransactionList = new ArrayList<>();
+        fruitTransactionList.add(fruitTransaction);
+        List<String> list = List.of("type;fruit;quantity", "r;apple;10");
+        Assert.assertEquals(fruitTransactionList, transactionParser.parse(list));
+    }
+
+    @Test
+    public void parse_incorrectOperation_notOk() {
+        TransactionParser transactionParser = new TransactionParserImpl();
+        List<String> list = List.of("type;fruit;quantity", "x;apple;5");
+        thrownRule.expect(RuntimeException.class);
+        thrownRule.expectMessage("Can't find Operation type for this String ");
+        transactionParser.parse(list);
+    }
+
+    @Test
+    public void parse_skipTitleRow_Ok() {
+        TransactionParser transactionParser = new TransactionParserImpl();
+        List<String> list = List.of("type;fruit;quantity", "b;apple;5");
+        System.out.println(transactionParser.parse(list));
```

Comment:
Each test should have some check, for example Assert. 
System.out.println is not good here